### PR TITLE
Feature/add players implementation

### LINKED ISSUE #10: REQ-009: Player Core System and Score Management
# Player Core System and Score Management Implementation

## Description
Create a doubly linked list system for players, including score management based on game difficulty and handling of duplicate nicknames.

### Data Structure
```c
typedef struct Player {
    int id;
    char nickname[MAX_STRING];
    float maxScore;
    struct Player* next;
    struct Player* prev;
} Player;
```

### Scoring System
* Points per correct answer:
  - Easy Mode: 0.5 points
  - Normal Mode: 1.0 points
  - God Mode: 2.0 points

### Required Functions
```c
Player* createPlayer(int id, char *nickname, float score);
void insertSortedPlayer(Player **head, Player *newPlayer);
Player* findPlayerByNickname(Player *head, char *nickname);
void updatePlayerIfHigherScore(Player **head, char *nickname, float newScore);
```

### Core Features
* Player Creation:
  - After difficulty selection
  - Unique ID assignment
  - Nickname input
* Score Management:
  - Calculate based on difficulty
  - Update highest scores
* Duplicate Handling:
  - Check existing nicknames
  - Keep highest score only
  - Update list order if needed
* Sorted List:
  - Maintain highest to lowest score order
  - Update order after score changes

## Acceptance Criteria
- [x] Player structure properly implemented
- [x] Creation function works correctly:
  - Proper ID assignment
  - Memory allocation
  - Initial score setup
- [x] Score calculation works:
  - Easy mode: 0.5 per question
  - Normal mode: 1.0 per question
  - God mode: 2.0 per question
- [x] Duplicate nickname handling:
  - Finds existing players
  - Updates if new score is higher
  - Maintains only highest score
- [x] Sorted insertion works:
  - Correct order by score
  - Proper link management
  - Handles all cases:
    * Empty list
    * Start/middle/end insertion
- [x] Memory management is proper
- [x] Double links maintained correctly

## Technical Notes
* Implement proper memory allocation/deallocation
* Handle all list edge cases
* Validate all inputs
* Maintain doubly linked integrity
* Document sorting algorithm
* Error handling implementation

## Integration Points
* Works with game difficulty selection
* Integrates with scoring system
* Maintains data consistency
* Prepares for persistence

### PR DESCRIPTION
**Pull request by J-P-T-0**
System for player implementation. Functionalities for player creation, scoring system and nickname creation. New pull request. New changes made to root feature/add players implementation in a string const for the game's questions. Revert of previous pull request, this is the new submission.

### CODE REVIEW

**Comment by J-P-T-0** [comment]
PR after revert of previous one

File: marathon_quiz.cpp [hunk 1]
```diff
@@ -19,6 +19,7 @@ typedef struct Player{
 	struct Player* prev;
 }Player;
 
+
```
Comment (gfdfranco) [inline marathon_quiz.cpp]: Please remove this symbol, and remove also the space

**Review by gfdfranco — CHANGES_REQUESTED** [review]

**Review by gfdfranco — APPROVED** [review]
LGTM

**Comment by J-P-T-0** [comment]
ok

**Review by Nohelany — APPROVED** [review]
LGTM

**Review by Mr-Mauxerr — APPROVED** [review]
LGTM

**Pull request merged**